syntax = "proto3";

package tt;

// Neighbors discovery service
service NeighborsDiscovery {
  rpc Greet (GreetRequest) returns (GreetReply) {}
  rpc Heartbeat (HeartbeatRequest) returns (HeartbeatReply) {}
}

// Neighbors chat service
service NeighborsChat {
  rpc Tell (TellRequest) returns (TellReply) {}
  rpc Narrate (stream NarrateRequest) returns (NarrateReply) {}
}

message GreetRequest {
  string nickname = 1;
  string identity = 2;
  string ipAddressAndPort = 3;
}

message GreetReply {
  string nickname = 1;
  string identity = 2;
  string ipAddressAndPort = 3;
}

message HeartbeatRequest {
  string identity = 1;
}

message HeartbeatReply {
  string identity = 1;
}

message TellRequest {
  string identity = 1;
  string message = 2;
}

message TellReply {
  string identity = 1;
}

message NarrateRequest {
  string identity = 1;
  string message = 2;
}

message NarrateReply {
  string identity = 1;
}
